### PR TITLE
Change doc references

### PR DESCRIPTION
**Pull request by yim-lee**
Together with https://github.com/apple/swift-argument-parser/pull/531, updating GitHub pages references to those hosted by Swift Package Index.

### CODE REVIEW

**Comment by yim-lee** [comment]
@rauhul This PR addresses [your comment](https://github.com/apple/swift-argument-parser/pull/531#issuecomment-1336080428) in https://github.com/apple/swift-argument-parser/pull/531

**Comment by rauhul** [comment]
LGTM, but @natecook1000 should probably verify that there's no other doc generation tools which include this string

**Comment by natecook1000** [comment]
@yim-lee Thanks for this! I've been keeping the docs parallel to the latest release, instead of the `main` branch. Does SPI support that approach?

File: CHANGELOG.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ The 1.2.0 release includes contributions from [allevato], [clayellis],
 [compnerd], [d-ronnqvist], [natecook1000], [randomeizer], and [rauhul].
 Thank you!
 
-[arrayparse-docs]: https://apple.github.io/swift-argument-parser/documentation/argumentparser/argumentarrayparsingstrategy
+[arrayparse-docs]: https://swiftpackageindex.com/apple/swift-argument-parser/main/documentation/argumentparser/argumentarrayparsingstrategy
```
Comment (natecook1000) [inline CHANGELOG.md:61]:
```suggestion
[arrayparse-docs]: https://swiftpackageindex.com/apple/swift-argument-parser/documentation/argumentparser/argumentarrayparsingstrategy
```

Comment (natecook1000) [inline CHANGELOG.md:61]:
```suggestion
[arrayparse-docs]: https://swiftpackageindex.com/apple/swift-argument-parser/documentation/argumentparser/argumentarrayparsingstrategy
```

File: README.md [hunk 2]
```diff
@@ -72,10 +72,10 @@ For guides, articles, and API documentation see the
 [library's documentation on the Web][docs] or in Xcode.
 
 - [ArgumentParser documentation][docs]
-- [Getting Started with ArgumentParser](https://apple.github.io/swift-argument-parser/documentation/argumentparser/gettingstarted)
-- [`ParsableCommand` documentation](https://apple.github.io/swift-argument-parser/documentation/argumentparser/parsablecommand)
+- [Getting Started with ArgumentParser](https://swiftpackageindex.com/apple/swift-argument-parser/main/documentation/argumentparser/gettingstarted)
+- [`ParsableCommand` documentation](https://swiftpackageindex.com/apple/swift-argument-parser/main/documentation/argumentparser/parsablecommand)
```
Comment (natecook1000) [inline README.md:76]:
```suggestion
- [Getting Started with ArgumentParser](https://swiftpackageindex.com/apple/swift-argument-parser/documentation/argumentparser/gettingstarted)
- [`ParsableCommand` documentation](https://swiftpackageindex.com/apple/swift-argument-parser/documentation/argumentparser/parsablecommand)
```

File: README.md [hunk 3]
```diff
@@ -72,10 +72,10 @@ For guides, articles, and API documentation see the
...
 
-[docs]: https://apple.github.io/swift-argument-parser/documentation/argumentparser/
+[docs]: https://swiftpackageindex.com/apple/swift-argument-parser/main/documentation/argumentparser
```
Comment (natecook1000) [inline README.md:78]:
```suggestion
[docs]: https://swiftpackageindex.com/apple/swift-argument-parser/documentation/argumentparser
```

**Review by natecook1000 — COMMENTED** [review]
Okay, I see — release support _is_ there, but SPI didn't build those docs because we added the `.spi.yml` file after `1.2.0` was released. That should be resolved when we do a `1.2.1` release, so I'll merge this change as part of that process.

In the meantime, let's drop the `main/` from the URLs so that SPI automatically forwards to the latest release. (You can see this behavior with swift-atomics here: https://swiftpackageindex.com/apple/swift-atomics/documentation/atomics)

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**